### ISSUE TITLE
Improvement: Allow OpenAPI Event Bus address to be specified for an entire API tag

### ISSUE THREAD

**Issue by InfoSec812** [vert-x3/issues#597]
#### Describe the feature

it occurs to me that the way in which RouterBuilder.mountServicesFromExtensions() functions may not be "optimal". Right now, each operation can have a different eventbus address, but in practice that makes no sense. Operations would be associated with a Tag/API grouping, and as such would likely use the same eventbus address for the whole Tag/API group. 

For example, when I instantiate my services and bind them, it's binding a whole API and not individual operations.
Seems like it would make more sense to have the evenbus address associated with the Tag definition in OpenAPI Spec. Kinda like:

```
---
openapi: 3.0.2
info:
  title: My App
  version: 1.0.0
  description: My Application
servers:
  ## SNIP . . .
tags:
  - name: system
    x-vertx-event-bus: services.system
  - name: bills
    x-vertx-event-bus: services.bills
  - name: income
    x-vertx-event-bus: services.income
  - name: cashflow
    x-vertx-event-bus: services.cashflow
```

I believe this could be implemented *in addition* to the existing functionality in order to maintain backward compatibility and allow for additional flexibility.

#### Use cases

This will enhance the ability of Vert.x to be the bases for low-code API implementations based on OpenAPI contracts.

#### Contribution

I will work with @pmlopes to help implement this improvement.

**Comment by pmlopes** [comment]
So, if understand it right, you would prefer to not see all the repetition we currently have:

```yaml
openapi: 3.0.0
info:
  version: 1.0.0
  title: Test spec for OpenAPI3ServiceProxiesTest
  license:
    name: Apache 2.0 License
paths:
  /testC:
    post:
      operationId: test_ -C_
      tags:
        - anotherTest
      x-vertx-event-bus:
        address: anotherAddress
        method: testC
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/Payload"
      responses:
        200:
          description: ok
  /testD:
    post:
      operationId: testD
      tags:
        - anotherTest
      x-vertx-event-bus:
        address: anotherAddress
        method: testD
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/Payload"
      responses:
        200:
          description: ok
components:
  schemas:
    Payload:
      type: object
      properties:
        hello:
          type: string
        name:
          type: string
      required:
        - hello
        - name
      additionalProperties: false
```

The

```yaml
x-vertx-event-bus:
  address: anotherAddress
  method: testC
```

And rely on a tag to pick that adress and compute the method from the `operationID`, right?

**Comment by pmlopes** [comment]
This means we need to change the function:

https://github.com/vert-x3/vertx-web/blob/master/vertx-web-openapi/src/main/java/io/vertx/ext/web/openapi/impl/OpenAPI3Utils.java#L134-L170

To allow the new behavior

**Comment by InfoSec812** [comment]
Yes, I think that would be a great option. Then from a generated class with `@WebApiServiceGen`, I could use `ServiceBinder` with that and just iterate over the tags to bind those classes to the event bus. Otherwise, I have to try to iterate over every operation in the OpenAPI structure. I will look at putting together a PR and tests for the changes. Thanks for the pointer to the function!
